docs/source/tutorials/describing_tasks.ipynb

```diff
@@ -1,20 +1,563 @@
 {
```

Comment:
<p>Note that when executing this on my VM, it fails (or just hangs, I did not wait for the timeout) I assume because I do not have 32 cores and 1 gpu available. The idea is to have a notebook that can execute on the read the docs virtual machine, on the docker container we provide, or just on any devel OS when compiling the documentation and/or executing pytest with nbmake.</p><p> </p><p>I know we said we do not want expose "fake_resources" to the users but I see no alternative and I do not think at this point is too much a big deal. We can just add a Waning box explaining the risk/limitation of using it. If you agree, the tutorial would write a suitable <code>$HOME/.radical/pilot/configs/resource_local.json</code> before executing this cell. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/radical-cybertools/radical.pilot/pull/2879/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Reply:
<p>The default <code>local.localhost</code> config has <code>fake_resources</code> set to True already, so my assumption was that this works out of the box on <em>any</em> localhost.  As such I don't think that a local config setting the value again would change things unfortunately.  I would like to  find out what happened here on your VM - any hint in the logs?</p>

Reply:
No hints, will have to debug. Did you try this on any other OS than yours? Maybe a container as per Mikhail's README?

Here another test I run:

```
(docs) mturilli@mosto:~/github/radical.pilot$ pytest --nbmake docs/source/tutorials/describing_tasks.ipynb
============================================================================== test session starts ===============================================================================
platform linux -- Python 3.10.6, pytest-7.2.2, pluggy-1.0.0
rootdir: /home/mturilli/github/radical.pilot
plugins: anyio-3.6.2, nbmake-1.4.1
collected 1 item

docs/source/tutorials/describing_tasks.ipynb F                                                                                                                             [100%]

==================================================================================== FAILURES ====================================================================================
________________________________________________ /home/mturilli/github/radical.pilot/docs/source/tutorials/describing_tasks.ipynb ________________________________________________
A cell timed out while it was being executed, after 300 seconds.
The message was: Cell execution timed out.
Here is a preview of the cell contents:
-------------------
['# configure reporter output ', "report = ru.Reporter(name='radical.pilot')", "report.title('Tutorial: Describing Tasks (RP version %s)' % rp.version)", '', '# create session and managers']
...
['', '# add the pilot to the task manager and wait for the pilot to become active', 'tmgr.add_pilots(pilot)', 'pilot.wait(rp.PMGR_ACTIVE)', "report.info('pilot state: %s' % pilot.state)"]
-------------------

------------------------------------------------------------------------------- Captured log call --------------------------------------------------------------------------------
ERROR    traitlets:client.py:845 Timeout waiting for execute reply (300s).
============================================================================ short test summary info =============================================================================
FAILED docs/source/tutorials/describing_tasks.ipynb:: - A cell timed out while it was being executed, after 300 seconds.
========================================================================= 1 failed in 308.45s (0:05:08) ==========================================================================
```

Stack:

```
$ radical-stack

  python               : /home/mturilli/ve/docs/bin/python3
  pythonpath           :
  version              : 3.10.6
  virtualenv           : /home/mturilli/ve/docs

  radical.analytics    : 1.20.1
  radical.entk         : 1.30.0
  radical.gtod         : 1.20.1
  radical.pilot        : 1.21.0
  radical.saga         : 1.21.0
  radical.utils        : 1.21.0
```

Maybe the same problem I have reported on #2858?

Reply:
I will test this as part of the PR of the full documentation.